features/schema.py

```diff
@@ -99,6 +105,57 @@ class Meta:
     day = WeekdayEnum(required=True)
 
 
+class FeatureFilter(django_filters.FilterSet):
+    """Contains the filters to use when retrieving Features."""
+
+    class Meta:
+        model = models.Feature
+        fields = [
+            "distance_lte",
+            "updated_since",
+            "tagged_with_any",
+            "tagged_with_all",
+            "category",
+        ]
+
+    distance_lte = DistanceFilter(
+        field_name="geometry",
+        lookup_expr="distance_lte",
+        label=_("Fetch features within a given distance from the given geometry"),
+    )
+    updated_since = django_filters.IsoDateTimeFilter(
+        method="filter_updated_since",
+        label=_("Fetch features that have changed since specified timestamp"),
+    )
+    tagged_with_any = StringListFilter(
+        method="filter_tagged_with_any",
+        label=_("Fetch features tagged with any of the specified tags (ids)"),
+    )
+    tagged_with_all = StringListFilter(
+        method="filter_tagged_with_all",
+        label=_("Fetch features tagged with all of the specified tags (ids)"),
+    )
+    category = StringListFilter(
+        method="filter_category", label=_("Fetch features from included categories")
+    )
+
+    def filter_updated_since(self, queryset, name, value):
+        return queryset.filter(
+            Q(overrides__modified_at__gt=value) | Q(mapped_at__gt=value)
+        )
+
+    def filter_tagged_with_any(self, queryset, name, value):
+        return queryset.filter(tags__in=value).distinct()
+
+    def filter_tagged_with_all(self, queryset, name, value):
+        for v in value:
```

Comment:
I wonder if ArrayField would be more appropriate for tags. Then it could be possible to filter with contains https://docs.djangoproject.com/en/3.0/ref/contrib/postgres/fields/#contains which provides a bit of performance advantage, when there are many tags to filter. This might make it though to our performance story.

Reply:
@charn great job!

Reply:
Maybe, though I would first try to measure and pinpoint the sources for the perceived performance issues.

Reply:
of course this cannot underlie the current performance issues, as filters just made it there ) that is just one more thing that might be worth looking at.